src/foam/nanos/crunch/ruler/UCJDataExpiryRule.js

```diff
@@ -57,12 +59,10 @@
           name: 'applyAction',
           javaCode: `
             var ucj = (UserCapabilityJunction) obj;
-            ((RenewableData) ucj.getData())
-              .setExpiry(Date.from(LocalDate.now()
-                .plusDays(((UCJDataExpiryRule) rule).getExpiredIn())
-                .atStartOfDay(ZoneId.systemDefault())
-                .toInstant()
-              ));
+            var self = (UCJDataExpiryRule) rule;
+            ucj.setExpiryPeriod(self.getExpiryPeriod());
+            ucj.setExpiryPeriodTimeUnit(self.getExpiryPeriodTimeUnit());
+            ucj.setExpiry(ucj.calculateDate(x, null, ucj.getExpiryPeriod(), ucj.getExpiryPeriodTimeUnit()));
```

Comment:
i think this rule is for setting the expiry of the ucj data instead of ucj

Reply:
So you think we can just drop this rule?

Reply:
on second thought i think this also makes sense. It looks like the ucj.data.expiry set previously was used to configure ucj.expiry and now we're just configureing the ucj expiry directly

Reply:
looks like this is used only for riskrating at the moment and it's set to configure a certain expiry on the riskrating ucj based on the riskrating rank in the data..